### PR TITLE
middleware-1.3.40

### PR DESCRIPTION
**Pull request by Bluette-C-Riviere**

### CODE REVIEW

File: middleware/middleware-1.3.40.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+We have made possible to configure the timeout duration for EF queue to perform a migration. In some cases when performing a migration for a big EF queue the default timeout was not long enough leading to some errors.
+### Affected packages:
+- _fiskaltrust.Middleware.Queue.EF v1.3.40_
+
+## Feature: TAR export modi for the queue
+We have added several TAR export modi for the queue for the TAR-File export during the daily closing to avoid a growth of the database due to open transactions. We noticed that more and more database size were increasing leading to issues in terms of perfomance and data upload.To solve this issue we implemented a queue parameter allowing to configure different modi for the TAR-File export:
+- all, this modi enables the automatic TAR-File export from the TSE on the queue
+- none, this modi disables the automatic TAR-File export
+- erased, this modi works in a way that TAR-files are only exported and saved when they can be deleted from the TSE.
```
Comment (volllly) [inline middleware/middleware-1.3.40.md:46]:
```suggestion
- all, this mode enables the automatic TAR-File export from the TSE on the queue
- none, this mode disables the automatic TAR-File export
- erased, this mode works in a way that TAR-files are only exported and saved when they can be deleted from the TSE. 
```

**Review by volllly — APPROVED** [review]

File: middleware/2022-10-04-1.3.40.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+authors: poscreator
+slug: middleware/1.3.40
+tags: [Middleware, Germany]
+---
+
+# fiskaltrust.Middleware 1.3.40 (Germany)
+_october 04, 2022_
+
+In this version of the Middleware, we have added a SCU parameter to configure the Heap memory of the FCC, we also have added a queue parameter to configure the migrations timeout, we have modified the DSFINV-K export to exclude non pos-receipts lowering memory consumption, we have created some modi for the TAR-Export and we have fixed bugs regarding failing TAR-Export using a Fiskaly SCU and a bug concerning the indexes creation.
+
```
Comment (TSchmiedlechner) [inline middleware/2022-10-04-1.3.40.md:11]: Please add the following snippet after the intro to "cut" the short description:

```suggestion
<!--truncate-->
```

File: middleware/2022-10-04-1.3.40.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
+authors: poscreator
+slug: middleware/1.3.40
+tags: [Middleware, Germany]
+---
+
+# fiskaltrust.Middleware 1.3.40 (Germany)
```
Comment (TSchmiedlechner) [inline middleware/2022-10-04-1.3.40.md:7]:
```suggestion
# Middleware 1.3.40 (Germany)
```

File: middleware/2022-10-04-1.3.40.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+_october 04, 2022_
```
Comment (TSchmiedlechner) [inline middleware/2022-10-04-1.3.40.md:8]: Not required anymore, please delete

File: middleware/2022-10-04-1.3.40.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+### Affected packages:
+- _fiskaltrust.Middleware.Queue.EF v1.3.40_
+- _fiskaltrust.Middleware.Queue.MySQL v1.3.40_
+- _fiskaltrust.Middleware.Queue.SQLite v1.3.40_
+
+## Feature: TAR export modi for the queue
+We have added several TAR export modi for the queue for the TAR-File export during the daily closing to avoid a growth of the database due to open transactions. We noticed that more and more database size were increasing leading to issues in terms of perfomance and data upload.To solve this issue we implemented a queue parameter allowing to configure different modi for the TAR-File export:
+- all, this mode enables the automatic TAR-File export from the TSE on the queue
+- none, this mode disables the automatic TAR-File export
+- erased, this mode works in a way that TAR-files are only exported and saved when they can be deleted from the TSE.
```
Comment (TSchmiedlechner) [inline middleware/2022-10-04-1.3.40.md:49]: + Affected packages, maybe?

Comment (Bluette-C-Riviere) [reply middleware/2022-10-04-1.3.40.md:49]: packages added

**Review by volllly — APPROVED** [review]

**Pull request merged**